config-ui/src/components/blueprints/GitlabProjectsSelector.jsx

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import React, { useEffect, useState } from 'react'
+import { Checkbox, Intent, MenuItem, } from '@blueprintjs/core'
+import { MultiSelect } from '@blueprintjs/select'
+
+const GitlabProjectsSelector = (props) => {
+  const {
+    fetchGitlabProjects = () => [],
+    isFetchingGitlab = false,
+    configuredConnection,
+    placeholder = 'Select Projects',
+    items = [],
+    selectedItems = [],
+    activeItem = null,
+    disabled = false,
+    isLoading = false,
+    isSaving = false,
+    onItemSelect = () => {},
+    onRemove = () => {},
+    onClear = () => {},
+    itemRenderer = (item, { handleClick, modifiers }) => (
+      <MenuItem
+        active={modifiers.active}
+        disabled={
+          selectedItems.find(i => i?.id === item?.id)
+        }
+        key={item.value}
+        onClick={handleClick}
+        text={
+          selectedItems.find(i => i?.id === item?.id)
+            ? (
+              <>
+                <input type='checkbox' checked readOnly /> {item?.title}
+              </>
+              )
+            : (
+              <span style={{ fontWeight: 700 }}>
+                <input type='checkbox' readOnly /> {item?.title}
+              </span>
+              )
+        }
+        style={{
+          marginBottom: '2px',
+          fontWeight: items.includes(item) ? 700 : 'normal',
+        }}
+      />
+    ),
+    tagRenderer = (item) => item?.title,
+  } = props
+
+  const [query, setQuery] = useState('')
+  const [onlyQueryMemberRepo, setOnlyQueryMemberRepo] = useState(true)
```

Comment:
This prop being named `isPrivate` would be more fitting.

```suggestion
  const [isPrivate, setIsPrivate] = useState(true)
```

Reply:
No, if you join in a public repo. it also will be queried.